Extension/src/Debugger/configurationProvider.ts

```diff
@@ -23,8 +85,59 @@ abstract class CppConfigurationProvider implements vscode.DebugConfigurationProv
     /**
 	 * Returns a list of initial debug configurations based on contextual information, e.g. package.json or folder.
 	 */
-    provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken): vscode.ProviderResult<vscode.DebugConfiguration[]> {
-        return this.provider.getInitialConfigurations(this.type);
+    async provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken): Promise<vscode.DebugConfiguration[]> {
+        const buildTasks: vscode.Task[] = await getBuildTasks();
+        if (!buildTasks.length) {
+            return Promise.resolve(this.provider.getInitialConfigurations(this.type));
+        }
+        const defaultConfig: vscode.DebugConfiguration = this.provider.getInitialConfigurations(this.type).find(config => {
+            return config.name === "(gdb) Launch" || config.name === "(lldb) Launch";
+        });
+        assert(defaultConfig, "Could not find default debug configuration.");
+
+        const platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+        const platform: string = platformInfo.platform;
+
+        // Generate new configurations for each build task.
+        // Generating a task is async, therefore we must *await* *all* map(task => config) Promises to resolve.
+        let configs: vscode.DebugConfiguration[] = await Promise.all(buildTasks.map<Promise<vscode.DebugConfiguration>>(async task => {
+            const definition: BuildTaskDefinition = task.definition as BuildTaskDefinition;
+            const compilerName: string = path.basename(definition.compilerPath);
+            const compilerDirname: string = path.dirname(definition.compilerPath);
+
+            let newConfig: vscode.DebugConfiguration = Object.assign({}, defaultConfig); // Copy enumerables and properties
+
+            newConfig.name = compilerName + " Build and Debug active file";
+            newConfig.preLaunchTask = task.name;
+            newConfig.externalConsole = false;
+            const exeName: string = path.join("${fileDirname}", "${fileBasenameNoExtension}");
+            newConfig.program = platform === "win32" ? path.join(exeName, ".exe") : exeName;
+
+            let debuggerName: string;
+            if (compilerName.startsWith("clang")) {
+                newConfig.MIMode = "lldb";
+                const suffix: string = compilerName.substr(compilerName.indexOf("-"));
```

Comment:
I don't understand what this suffix is supposed to be. It's turning "clang" into "g" and then lldb into lldbg.

Reply:
I think it's supposed to be the `6.0` after the `-` in `clang-6.0`.  Problem is that the user could just be using `clang`.